homeassistant/components/varta_storage/__init__.py

```diff
@@ -0,0 +1,252 @@
+            "device_class": "energy",
+            "state_class": "total",
+            "native_unit_of_measurement": ENERGY_KILO_WATT_HOUR,
+        }
+        thisdict[12] = {
+            "name": "VARTA total grid to home",
+            "native_value": result.grid_to_home,
+            "device_class": "energy",
+            "state_class": "total",
+            "native_unit_of_measurement": ENERGY_WATT_HOUR,
+        }
+        thisdict[13] = {
+            "name": "VARTA total home to grid",
+            "native_value": result.home_to_grid,
+            "device_class": "energy",
+            "state_class": "total",
+            "native_unit_of_measurement": ENERGY_WATT_HOUR,
+        }
+        thisdict[14] = {
+            "name": "VARTA invertert total charged",
+            "native_value": result.inverter_total_charged,
+            "device_class": "energy",
+            "state_class": "total",
+            "native_unit_of_measurement": ENERGY_WATT_HOUR,
+        }
+        thisdict[15] = {
+            "name": "VARTA invertert total discharged",
+            "native_value": result.inverter_total_discharged,
+            "device_class": "energy",
+            "state_class": "total",
+            "native_unit_of_measurement": ENERGY_WATT_HOUR,
+        }
+        thisdict[16] = {
+            "name": "VARTA charge cycle counter",
+            "native_value": result.charge_cycle_counter,
+            "device_class": "None",
+            "state_class": "None",
+            "native_unit_of_measurement": None,
+        }
+        thisdict[17] = {
+            "name": "VARTA hours until filter maintenance",
+            "native_value": result.hours_until_filter_maintenance,
+            "device_class": "None",
+            "state_class": "None",
+            "native_unit_of_measurement": TIME_HOURS,
+        }
+        thisdict[18] = {
+            "name": "VARTA fan",
+            "native_value": result.fan,
+            "device_class": "None",
+            "state_class": "None",
+            "native_unit_of_measurement": None,
+        }
+        thisdict[19] = {
+            "name": "VARTA main",
+            "native_value": result.main,
+            "device_class": "None",
+            "state_class": "None",
+            "native_unit_of_measurement": None,
+        }
+
+        return thisdict
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="sensor",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=1),
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    # Forward the setup to the sensor platform.
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, "sensor")
+    )
+
+    return True
+
+
+async def options_update_listener(
```

Comment:
```suggestion
async def async_reload_entry(
```

Reply:
Done by @MSiegrist , thank you @frenck for this recommendation